Updating guidance on pipeline parameters use with dynamic config (#9627)
circleci/circleci-docs: docs/guides/modules/orchestrate/pages/dynamic-config.adoc

Some constraints on continuing pipelines from a `setup: true` configuration are as follows:

- The setup configuration must be `version: 2.1`.
- The setup configuration must only allow one workflow to run in each pipeline. For example:
** Have only one workflow configured.
** Use logic conditions to ensure only one workflow is run for each pipeline.
- A pipeline can only be continued once. A pipeline cannot be continued with another setup configuration. Workflow reruns will fail as part of this restriction. Rather than rerun a setup workflow you can trigger a new pipeline.
- A pipeline can only be continued within six hours of its creation.

> - Pipeline parameters submitted at continuation time cannot overlap with pipeline parameters submitted at trigger (setup) time. When xref:triggers-overview.adoc#run-a-pipeline-using-the-api[triggering a pipeline via the API], you cannot set a pipeline parameter to another value than the default one declared in the setup configuration if that pipeline parameter is also declared in the continuation configuration.

Reviewer: ```suggestion
- Pipeline parameters submitted at continuation time cannot overlap with pipeline parameters submitted at trigger (setup) time. When xref:triggers-overview.adoc#run-a-pipeline-using-the-api[triggering a pipeline via the API], you cannot set a pipeline parameter to a value other than the default value declared in the setup configuration if that pipeline parameter is also declared in the continuation configuration.
```

> - Pipeline parameters declared in the setup configuration and pipeline parameters declared in the continuation configuration are independent. If you declare a pipeline parameter with the same name and type in both the setup and continuation configuration files, but with different default values, the default values will apply respectively to each configuration.

Author: Thanks @briceicle. Indeed, that sentence is somewhat contradicting the statement in the previous point (line 57). But we also wanted to highlight the behaviour we've observed. I'll keep the rest of the point (line 58), but I'm removing that first sentence.

> - You can set the value of a pipeline parameter declared in the continuation configuration to a different value than the default one. To do so, you need to specify the `parameters` parameter to pass both the parameter's name and value when calling the `continue` job or command of the link:https://circleci.com/developer/orbs/orb/circleci/continuation[continuation orb].

Reviewer: ```suggestion
- You can set the value of a pipeline parameter declared in the continuation configuration to a value  other than the default. To do so, specify the parameter's name and value when calling the `continue` job or command from the link:https://circleci.com/developer/orbs/orb/circleci/continuation[continuation orb] using the `parameters` parameter.
```

Some constraints on continuing pipelines from a `setup: true` configuration are as follows:

- The setup configuration must be `version: 2.1`.
- The setup configuration must only allow one workflow to run in each pipeline. For example:
** Have only one workflow configured.
** Use logic conditions to ensure only one workflow is run for each pipeline.
- A pipeline can only be continued once. A pipeline cannot be continued with another setup configuration. Workflow reruns will fail as part of this restriction. Rather than rerun a setup workflow you can trigger a new pipeline.
- A pipeline can only be continued within six hours of its creation.
- When a pipeline parameter is declared in both the setup and the continuation configurations, the pipeline will error if both the following conditions are met:

. The pipeline is triggered via the API.

> . The pipeline parameter's value specified in the API request is different from the default value specified in the setup configuration.

Author: @SashaLsnko, yes. That would result in an error.

=== Use path filtering for monorepos

In a link:https://circleci.com/blog/monorepo-dev-practices/[monorepo] setup:

* Set up path filtering to set pipeline parameters and continuation configuration sources based on the location of changes in your repository.
* You can use path filtering to avoid all your microservices/sub-projects being put through the entirety of your build, test, and deployment processes when any single update occurs.

For example, you may have multiple projects within your repository (`project_1`, `project_2`, `project_3`). +
You can use path filtering to only run tests and deployment for `project_1` if change happened in the `project_1` file set. The link:https://circleci.com/developer/orbs/orb/circleci/path-filtering[path filtering orb] is available to help simplify your configuration further.

For a full example of path filtering, see the xref:using-dynamic-configuration.adoc#execute-specific-workflows-or-steps-based-on-which-files-are-modified[Dynamic configuration how-to guide].

// #DIAGRAM OF PATH FILTERING FOR MONOREPO SETUP#

=== Use config splitting for microservices

> For projects that consist of multiple modules, you may require a separate job/workflow to run for each module when changes are introduced. +

Reviewer: I find the formatting using + strange to look at. Can we just have either separate paragraphs, one big paragraph or bullet list sort of thing?
